tests/conftest.py

```diff
@@ -1596,8 +1621,28 @@ def verify_new_core_dumps(duthost):
     new_core_dumps = list(new_core_dumps)
 
     if new_core_dumps:
-        pytest.fail("Core dumps found. Expected: %s Found: %s. Test failed. New core dumps: %s" % (len(pre_existing_cores),\
-            len(cur_cores), new_core_dumps))
+        reload = True
+        logger.info("Core dumps found. Expected: %s Found: %s. New core dumps: %s" % (len(pre_existing_cores), len(cur_cores), new_core_dumps))
+
+    # Check if the running config is different from the golden config
+    if duthost.stat(path=CONFIG_DB_BAK).get('stat', {}).get('exists', False):
+        result = duthost.command("sonic-cfggen -d --print-data")
+        running_config = json.loads(result['stdout'])
+
+        result = duthost.command("cat {}".format(CONFIG_DB_BAK))
+        config_db_bak = json.loads(result['stdout'])
+
+        if compare_config(running_config, config_db_bak):
+            reload = True
+    else:
+        logger.info("There is no config_db.json.bak on host %s" % duthost.hostname)
+        return
+
+    # Reload with golden config
+    if reload:
+        logger.info("Reload with bak config")
+        duthost.command("cp {} {}".format(CONFIG_DB_BAK, CONFIG_DB))
+        config_reload(duthost)
```

Comment:
If reload is performed, this means that something went wrong. Then it would be better to fail the test after reload is done.

Since this fixture is doing multiple checks (core dump and config db). Suggest to record the check results in a dict. Then based on the check results to do recovery (config reload in this case). After config reload is done, still need to fail the test. Otherwise, the failures would not get human attention.